daemon/config/config_linux.go

```diff
@@ -108,14 +109,12 @@ func (conf *Config) GetInitPath() string {
 	return DefaultInitBinary
 }
 
-// LookupInitPath returns an absolute path to the "docker-init" binary by searching relevant "libexec" directories (per FHS 3.0 & 2.3) followed by PATH
-func (conf *Config) LookupInitPath() (string, error) {
-	binary := conf.GetInitPath()
+func lookupBinPath(binary string) (string, error) {
```

Comment:
Even though it's a private function, IMO we should copy most of the `LookupInitPath` godoc so the intent is still clear; maybe something like: :innocent:

```suggestion
// lookupBinPath returns an absolute path to the provided binary by searching relevant "libexec" locations (per FHS 3.0 & 2.3) followed by PATH
func lookupBinPath(binary string) (string, error) {
```